### PR TITLE
[Bug] Integrations-Pr Command (Elastic-Package Linting and Version Adjustments)

### PR DESCRIPTION
**Pull request by terrancedejesus**
## Issues
* https://github.com/elastic/detection-rules/issues/2008

## Summary
The `integrations-pr` command in devtools.py should be reviewed for the following:

- elastic_package references where linting is called (bug)

### CODE REVIEW

**Comment by terrancedejesus** [comment]
Output from using this command on the latest release for 7.16.4.

```
Already on 'main'
From github.com:elastic/integrations
 * branch                main       -> FETCH_HEAD
Switched to a new branch 'detection-rules/7.16.4-a7411d05'
2022/08/16 13:26:16  WARN CommitHash is undefined, in both /Users/tdejesus/.elastic-package/version and the compiled binary, config may be out of date.
2022/08/16 13:26:16  INFO New version is available - v0.59.0. Download from: https://github.com/elastic/elastic-package/releases/tag/v0.59.0
Format the package
Done
Enumerating objects: 1262, done.
Counting objects: 100% (1262/1262), done.
Delta compression using up to 10 threads
Compressing objects: 100% (638/638), done.
Writing objects: 100% (640/640), 191.20 KiB | 7.35 MiB/s, done.
Total 640 (delta 600), reused 1 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (600/600), completed with 572 local objects.
remote: 
remote: Create a pull request for 'detection-rules/7.16.4-a7411d05' on GitHub by visiting:
remote:      https://github.com/elastic/integrations/pull/new/detection-rules/7.16.4-a7411d05
remote: 
To github.com:elastic/integrations.git
 * [new branch]          detection-rules/7.16.4-a7411d05 -> detection-rules/7.16.4-a7411d05
PR created:
https://github.com/elastic/integrations/pull/4010
2022/08/16 13:26:47  WARN CommitHash is undefined, in both /Users/tdejesus/.elastic-package/version and the compiled binary, config may be out of date.
2022/08/16 13:26:47  INFO New version is available - v0.59.0. Download from: https://github.com/elastic/elastic-package/releases/tag/v0.59.0
Format the package
Done
2022/08/16 13:26:47  WARN CommitHash is undefined, in both /Users/tdejesus/.elastic-package/version and the compiled binary, config may be out of date.
2022/08/16 13:26:47  INFO New version is available - v0.59.0. Download from: https://github.com/elastic/elastic-package/releases/tag/v0.59.0
Lint the package
Done
Enumerating objects: 9, done.
Counting objects: 100% (9/9), done.
Delta compression using up to 10 threads
Compressing objects: 100% (5/5), done.
Writing objects: 100% (5/5), 450 bytes | 450.00 KiB/s, done.
Total 5 (delta 4), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (4/4), completed with 4 local objects.
To github.com:elastic/integrations.git
   644eef8b3..b43e2d784  detection-rules/7.16.4-a7411d05 -> detection-rules/7.16.4-a7411d05
```

**Comment by terrancedejesus** [comment]
For now, we can address the bug in the code that stops it from normal execution and then focus on the enhancements at another time. The following is being addressed in these code adjustments.

- extra linting line caused errors with `elastic-package` go package
- split the package version string from packages.yaml in case the `-dev.0` is left in there. If it does not exist, the semantic version is still pulled correctly. Based on the step-by-step documentation, this value is changes to the release version and imported so as long as the process is followed as usual no errors should occur.

The following are enhancements we want to do:

- Update integrations manifest and code to include `security_detection_engine`. This will be needed in the integrations manifest to reference what the latest version is based on the stack version. For example, if on 7.16, load integrations manifest, filter for key `security_detection_engine` and sort version, returning the latest. We can then include a click prompt, comparison logic and an error if the expected does not match the input.

File: detection_rules/devtools.py [hunk 1]
```diff
@@ -591,7 +590,7 @@ def elastic_pkg(*args):
     click.echo("PR created:")
     click.echo(pr.html_url)
 
-    # replace the changelog entry with the actual PR link
+    # # replace the changelog entry with the actual PR link
```
Comment (terrancedejesus) [inline detection_rules/devtools.py:593]:
```suggestion
    # replace the changelog entry with the actual PR link
```

**Review by Mikaayenson — APPROVED** [review]
LGTM

File: detection_rules/devtools.py [hunk 2]
```diff
@@ -471,7 +471,7 @@ def integrations_pr(ctx: click.Context, local_repo: str, token: str, draft: bool
 
     local_repo = os.path.abspath(local_repo)
     stack_version = Package.load_configs()["name"]
-    package_version = Package.load_configs()["registry_data"]["version"]
+    package_version = Package.load_configs()["registry_data"]["version"].split("-")[0]
```
Comment (terrancedejesus) [inline detection_rules/devtools.py:474]:
```suggestion
    package_version = Package.load_configs()["registry_data"]["version"]
```

**Review by brokensound77 — APPROVED** [review]

**Pull request merged**